activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -125,24 +125,31 @@ def migration_context # :nodoc:
 
       def schema_migration # :nodoc:
         @schema_migration ||= begin
-                                conn = self
-                                spec_name = conn.pool.db_config.spec_name
-                                name = "#{spec_name}::SchemaMigration"
-
-                                Class.new(ActiveRecord::SchemaMigration) do
-                                  define_singleton_method(:name) { name }
-                                  define_singleton_method(:to_s) { name }
-
-                                  connection_handler.connection_pool_names.each do |pool_name|
-                                    if conn.pool == connection_handler.retrieve_connection_pool(pool_name)
-                                      self.connection_specification_name = pool_name
-                                      break
-                                    end
+                                superclass = find_connection_handler
+                                if superclass == ActiveRecord::Base
+                                  ActiveRecord::SchemaMigration
```

Comment:
It doesn't *have* to, but it makes it so that `ActiveRecord::Base.connection.schema_migration == ActiveRecord::SchemaMigration`, which is nicer than getting an anonymous class, and better for backward compat.

All this is very temporary though, I really don't like the current `schema_migration` implementation, it's just hard to do better for now.